### ISSUE TITLE
LINCS hit query

### ISSUE THREAD

**Issue by gwaygenomics** [broadinstitute/lincs-profiling-comparison#14]
@michaelbornholdt presented an analysis on querying "hits" during profiling checkin today.

I'd love to be able to include this analysis in the LINCS complementarity paper. Michael estimated that his analysis would take ~2 hours. 

## Specification

- Perform hit analysis in L1000 data
- Perform hit analysis in Cell Painting data (CellProfiler features)

## Input data

- L1000 profiles [here](https://github.com/broadinstitute/lincs-profiling-comparison/blob/1d188ecbf38923a45d1c64f8cecb01ab04566491/1.Data-exploration/Profiles_level4/L1000/L1000_lvl4_cpd_replicate_datasets/L1000_level4_cpd_replicates.csv.gz)
- Cell Painting profiles [here](https://github.com/broadinstitute/lincs-profiling-comparison/blob/1d188ecbf38923a45d1c64f8cecb01ab04566491/1.Data-exploration/Profiles_level4/cell_painting/cellpainting_lvl4_cpd_replicate_datasets/cp_level4_cpd_replicates.csv.gz)

## Output

I think I need to understand the "hit" analysis better. Given a compound, you're asking if you match another replicate as the top hit? So essentially the output is a ranked list of matches per compound and whether or not they map to the same category?

If so, then can you output the following data frame?

| target_compound | compound_to_match | rank | same_replicate | same_MOA |
| :----------------: | :---------------------: | :---: | :-------------: | :-----------: |
| X | Y | 1 | True | True |
| X | Z | 2 | False | True |

Let's iterate on the final output specifications if my understanding above is limited in any way.

A couple preliminary figures and statistics would also be helpful.

## Motivation

- I think it'll be great to include this in the LINCS complementarity project b/c it aligns with one of the LINCS goals of being able to query signatures and infer function by guilt-by-association
- It will be a cool analysis to compare the efficient-net features vs. CellProfiler features (even though we will not include efficient net in this paper)
- If you perform this analysis and it makes it into the paper, we would offer you authorship (which would also require you to review the final submission and stand by its claims)

**Comment by michaelbornholdt** [comment]
Thanks for setting up this issue. I agree with the motivation and the authorship, if we include this. 

I can definitely give you that output. But it will take a bit longer to build that df. You proposed df also has a lot more information then needed for the simple histogram. 
But overall its best because I will want to put this onto cyto eval anyways.

**Comment by michaelbornholdt** [comment]
We should also add the MOAs of both compounds to that df since that will allow for more MOA focussed plots instead of counting the compounds

**Comment by gwaygenomics** [comment]
Sounds good! I am aiming for a July 1 submission, so in order for us to include it, I will need it before then.

Two additional points that you might want to consider:

- One path to getting this on cytominer-eval is to use this analysis as a test case. Apply it without the constraints of a package, learn how you need to wrangle the input and output data, what info is required, etc. and only after you've validated the output, code the general case.
- The MOA information can be included in this output for sure, I was thinking not to since it can be acquired with a simple join on compound name, and it will substantially increase data size. I agree it is nice to have though, so definitely up to u.

Thanks!

**Comment by michaelbornholdt** [comment]
Yea good points! 
Not very familiar with test-driven development but I could try it :)

**Comment by michaelbornholdt** [comment]
@gwaygenomics 
The data you linked up there is level3. I just require level5 data. I misspoke in the meeting! 

Can you point me to which level5 I should be using

**Comment by gwaygenomics** [comment]
level 5 links:
[Cell Painting](https://github.com/broadinstitute/lincs-profiling-comparison/blob/1d188ecbf38923a45d1c64f8cecb01ab04566491/1.Data-exploration/Consensus/cell_painting/moa_sizes_consensus_datasets/modz_consensus_data.csv)
[L1000](https://github.com/broadinstitute/lincs-profiling-comparison/blob/1d188ecbf38923a45d1c64f8cecb01ab04566491/1.Data-exploration/Consensus/L1000/moa_sizes_consensus_datasets/modz_level5_data.csv)

Thanks!

**Comment by gwaygenomics** [comment]
@michaelbornholdt produced the analysis here: https://github.com/broadinstitute/neural-profiling/issues/2#issuecomment-872433897

I will ingest the output files in this repo for visualization
